draft-ietf-tls-tls13.md

```diff
@@ -2242,12 +2267,15 @@ extension type (Supported Group Extension):
 The client MUST supply a "supported_groups" extension containing at
 least one group for each key exchange algorithm (currently
 DHE and ECDHE) for which it offers a cipher suite.
-If the client does not supply a "supported_groups" extension with a
-compatible group, the server MUST NOT negotiate a cipher suite of the
-relevant type.  For instance, if a client supplies only ECDHE groups,
-the server MUST NOT negotiate finite field Diffie-Hellman.  If no
-acceptable group can be selected across all cipher suites, then the
-server MUST generate a fatal "handshake_failure" alert.
+Servers receiving a TLS 1.3 ClientHello without this extension yet
+offering either DHE or ECDHE cipher suites MUST send a
+"missing_extension" alert message and close the connection.
```

Comment:
I disagree with that. If the extension is mandatory to send, then a hello without it is essentially malformed. Not requiring it to check if it's not negotiating (EC)DHE at the moment might be ok, if that's what you mean.